lib/host_kafka.py

```diff
@@ -0,0 +1,35 @@
+import socket
+from contextlib import closing
+
+from app.config import Config
+from app.environment import RuntimeEnvironment
+from app.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+def _any_bootstrap_server_connects(kafka_socket, servers) -> bool:
+    if not servers:
+        config = Config(RuntimeEnvironment.SERVICE)
+        servers = [config.bootstrap_servers] if isinstance(config.bootstrap_servers, str) else config.bootstrap_servers
+
+    try:
+        for server in servers:
+            try:
+                host, port = server.split(":")
+                new_addr = (host, int(port))
+
+                # connect_ex() returns zero when the socket is open and accessible
+                errcode = kafka_socket.connect_ex(new_addr)
+                if not errcode:
+                    return True
+
+            except ValueError as ve:
+                logger.error(f"Invalid server address: {str(ve)}")
+    except ValueError as ve:
+        logger.error(f"No servers available: {str(ve)}")
+
+
+def kafka_available(servers=None):
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as kafka_socket:
+        return _any_bootstrap_server_connects(kafka_socket, servers)
```

Comment:
This goes along with the other comments in this review about changing how exceptions are handled. I think it would make sense to have the KafkaError raised here instead of where this is called. Then callers can chose to handle the exception as they see fit.

```suggestion
def kafka_available(servers=None):
    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as kafka_socket:
        if not _any_bootstrap_server_connects(kafka_socket, servers):
            raise KafkaError("No Kafka server available")
```